application/utils/noise_filter/sanitize.py

```
@@ -0,0 +1,142 @@
+    ("ﬂ", "fl"),  # ﬂ
+]
+
+# word-\nword continuation from PDF line-wrapping. Matches "foo-\nbar" -> "foobar".
+_HYPHEN_BREAK_RE: re.Pattern[str] = re.compile(r"(\w)-\n(\w)")
+
+# HTML/XML tags. <p>, </script>, <br/>, etc. Doesn't match `<` followed by space.
+_HTML_TAG_RE: re.Pattern[str] = re.compile(r"</?[a-zA-Z][^>]*>")
+
+# Zero-width characters that shift embedding spaces but render invisibly:
+# ZWSP (U+200B), ZWNJ (U+200C), ZWJ (U+200D), BOM/ZWNBSP (U+FEFF).
+_ZERO_WIDTH_RE: re.Pattern[str] = re.compile("[​‌‍﻿]")
```

Ruff (0.15.15)

[error] Invalid unescaped character zero-width-space, use "\u200B" instead

Replace with escape sequence

(PLE2515)